django_socio_grpc/request_transformer/grpc_internal_proxy.py

```diff
@@ -39,16 +46,103 @@ class GRPCInternalProxyResponse:
     """
 
     grpc_response: ResponseType
-    http_response: InternalHttpResponse = None
+    # INFO - AM - 25/07/2024 - grpc context is used to pass response header to client
+    grpc_context: ServicerContext
+    # INFO - AM - 01/08/2024 - http_response is created in post_init signals. Don't need to pass it in the constructor if defautl behavior wanted.
+    http_response: InternalHttpResponse | None = None
+    # INFO - AM - 01/08/2024 - headers is created in post_init signals. Don't need to pass it in the constructor if defautl behavior wanted.
+    headers: Optional["ResponseHeadersProxy"] = None
 
     def __post_init__(self):
         self.http_response = InternalHttpResponse()
+        self.headers = ResponseHeadersProxy(self.grpc_context, self.http_response)
 
     def __getattr__(self, attr):
+        """
+        This private method is used to correctly distribute the attribute access between the proxy, th grpc_response and the http_response
```

Comment:
Not sure about what you mean. If you ask why having both GRPCInternalProxyResponse and InternalHttpResponse instead of just one doing the work than answers are:
- It was originally like that
- It allow more flexibility about usage 
- It allow to potentially fix name conflicts 

If you think it's better to reduce code complexity by merging both class why not but we may want to discuss more about it and test it maybe in an other PR